coordinator/src/services/translation/signing.ts

```diff
@@ -0,0 +1,147 @@
+  })
+
+  fragments.pertinentLineItem.forEach(
+    lineItem => fragmentsToBeHashed.push({
+      pertinentLineItem: lineItem
+    })
+  )
+
+  return writeXmlStringCanonicalized({
+    FragmentsToBeHashed: {
+      Fragment: fragmentsToBeHashed
+    }
+  } as XmlJs.ElementCompact)
+}
+
+export function convertFragmentsToDisplayableFormat(fragments: Fragments): Display {
+  const prescriptionStartDate = fragments.time._attributes.value
+
+  const patient = fragments.recordTarget.Patient
+  const patientPerson = patient.patientPerson
+
+  const patientName =
+    joinCoreText(" ",
+      patientPerson.name[0].prefix,
+      patientPerson.name[0].given,
+      patientPerson.name[0].family
+    )
+  const patientDob = patientPerson.birthTime._attributes.value
+  const patientGender = getGenderFromCode(patientPerson.administrativeGenderCode)
+  const patientNhsNumber =  patient.id._attributes.extension
+  const patientAddress = joinCoreText(", ",
+    patient.addr[0].streetAddressLine,
+    patient.addr[0].postalCode
+  )
+
+  const prescriber = fragments.agentPerson
+  const prescriberPerson = prescriber.agentPerson
+  const prescribingOrg = prescriber.representedOrganization
+
+  const prescriberName = joinCoreText(" ",
+    prescriberPerson.name.prefix,
+    prescriberPerson.name.given,
+    prescriberPerson.name.family
+  )
+  const prescriberCode = prescriber.id._attributes.extension
+  const prescriberOrg = `${prescribingOrg.name._text} (${prescribingOrg.id._attributes.extension})`
+  const prescriberAddress = joinCoreText(", ",
+    prescribingOrg.addr.streetAddressLine,
+    prescribingOrg.addr.postalCode
+  )
+
+  const medication = fragments.pertinentLineItem
+    .map((medication) => {
+      return new DisplayMedication({
+        name: medication.product.manufacturedProduct.manufacturedRequestedMaterial.code._attributes.displayName,
+        quantity: medication.component.lineItemQuantity.quantity._attributes.value,
+        dose: medication.pertinentInformation2.pertinentDosageInstructions.value
+      })
+    })
+
+  const display: Display = {
+    prescriptionStartDate: prescriptionStartDate,
+    patientName: patientName,
+    patientDob: patientDob,
+    patientGender: patientGender,
+    patientNhsNumber: patientNhsNumber,
+    patientAddress: patientAddress,
+    prescriberName: prescriberName,
+    prescriberCode: prescriberCode,
+    prescriberOrg: prescriberOrg,
+    prescriberAddress: prescriberAddress,
+    medication: medication
+  }
+
+  return display
+}
+
+/* eslint-disable  @typescript-eslint/no-explicit-any */
+function joinCoreText(seperator: string, ...values: any[]) : string {
+  return values.map(unpackCoreText).filter(Boolean).join(seperator)
+}
+
+function unpackCoreText(value: core.Text | core.Text[]) : string {
+  if (isCoreTextArray(value)) {
+    return value.map(v => v._text).join(" ")
+  } else {
+    const text = value as core.Text
+    return text !== undefined ? text._text : ""
```

Comment:
if text is null or an empty object it will still try and access `text._text` which will throw an error - what you think about just writing `return text ? text._text : ""`?